### PR TITLE
refactor: convert to JS

### PR DESCRIPTION
**Pull request by mhmd-azeez**
This PR does two things:
1. Converts index.ts to index.js
2. Puts the wasm file in `dist/plugin.wasm` by default (this makes it consistent with what we are doing in xtp)

I have also forked this repo at https://github.com/extism/ts-pdk-template for the actual TS template

### CODE REVIEW

**Comment by mhmd-azeez** [comment]
See https://github.com/extism/cli/pull/85

**Review by bhelx — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
     "description": "",
     "main": "src/index.ts",
     "scripts": {
-      "build": "node esbuild.js && extism-js dist/index.js -i src/index.d.ts -o plugin.wasm"
+      "build": "node esbuild.js && extism-js dist/index.js -i src/index.d.ts -o dist/plugin.wasm"
```
Comment (bhelx) [inline package.json:7]: Let's also remove `typescript` from the devDependencies

File: package.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
     "description": "",
     "main": "src/index.ts",
```
Comment (zshipko) [inline package.json]:
```suggestion
    "main": "src/index.js",
```

Comment (mhmd-azeez) [reply package.json]: thanks!

File: package.json (hunk 1 above)
Comment (mhmd-azeez) [reply package.json:7]: good call!

**Pull request merged**